### PR TITLE
[WASI-NN] Update ggml

### PR DESCRIPTION
**Pull request by dm4**
- Update ggml library (#2802)
- Use gguf model for tests
- Enable `LLAMA_BLAS` by default

### CODE REVIEW

**Review by hydai — CHANGES_REQUESTED** [review]

File: LICENSE.spdx [hunk 1]
```diff
@@ -39,7 +39,7 @@ PackageFileName: ./thirdparty/ggml
 PackageHomePage: https://github.com/ggerganov/llama.cpp
 PackageOriginator: Georgi Gerganov
 PackageLicenseDeclared: MIT
-PackageDownloadLocation: git://github.com/ggerganov/llama.cpp.git@da39a3ee5e6b4b0d3255bfef95601890afd80709
```
Comment (hydai) [inline LICENSE.spdx:42]: The `PackageFileName` should be updated at the same time.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
Is it possible to know if the OpenBLAS is set correctly?

**Comment by dm4** [comment]
> Is it possible to know if the OpenBLAS is set correctly?

In cmake configure log:
```
-- BLAS found, Libraries: /usr/lib/x86_64-linux-gnu/libopenblas.so
```

or check the plugin binary dependency:
```
$ ldd ~/.wasmedge/plugin/libwasmedgePluginWasiNN.so | grep openblas
        libopenblas.so.0 => /lib/x86_64-linux-gnu/libopenblas.so.0 (0x00007f749cf88000)
```

**Comment by dm4** [comment]
Continue at the new PR #2831

**Pull request closed without merging**